include/exadg/postprocessor/time_control.h

```diff
@@ -0,0 +1,99 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+#define INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_TIME_CONTROL_H_
+
+/**
+ *This class provides information at which time steps a certain action should be triggered. It is
+ *typically used in postprocessing routines that shall not be invoked after every time step. The
+ *user can specify an evaluation interval in terms of number of time steps or an amount of time.
+ */
+
+// deal.ii
+#include <deal.II/base/conditional_ostream.h>
+
+// ExaDG
+#include <exadg/utilities/numbers.h>
+
+namespace ExaDG
+{
+struct TimeControlData
+{
+  TimeControlData();
+
+  bool         is_active;
+  double       start_time;
+  double       end_time;
+  double       trigger_interval;
+  unsigned int trigger_every_time_steps;
```

Comment:
I would argue that you have to use `types::time_step` here, both variables are combined with each other via mathematical operators (addition, modulo) in `needs_evaluation`. The current implementation allows to use different types, implying implicit type conversions. This could potentially become very ugly in terms of steady_timestep, invalid_timestep, etc. and the values choosing for these constants.

Reply:
I would argue that trigger_every_time_step is not a time_step but a number. And we are safe to do everything if we keep time_step also an unsigned int, which makes sense since a negative time_step can not exist. This is also true for the steady case and this is why I implemented the numbers::invalid_time_step in the first place.

Reply:
You argue that every_time_steps is a number (which would allow a negative value), but at the same time you argue with time_step, which cannot be negative. So I think this point is unclear so far?

Reply:
Its a positive number. Evaluating every -10th time-step does not make any sense.

Reply:
I do not want to dispute this. But for me this implies it has to be of type `time_step` also because you do calculations between both variables. Put differently, why do you introduce `types::time_step` here if you do not want to make use of the type in the sense that we might change the concrete type in the future, where everything still has to work in such a case.

Reply:
I can't think about any possible reason that this changes. Nevertheless:
[5e6d7a9](https://github.com/exadg/exadg/pull/262/commits/5e6d7a969014aa50aaba54a91cffe275f7145346)